src/Network/Request.php

```diff
@@ -1375,6 +1381,42 @@ public function setInput($input)
     }
 
     /**
+     * Update the request with a new request data element.
+     *
+     * Returns an updated request object. This method returns
+     * a *new* request object and does not mutate the request in-place.
+     *
+     * @param string $name The dot separated path to insert $value at.
+     * @param mixed $value The value to insert into the request data.
+     * @return self
+     */
+    public function withData($name, $value)
+    {
+        $copy = clone $this;
+        $copy->data = Hash::insert($copy->data, $name, $value);
+
+        return $copy;
+    }
```

Comment:
yuck. That's going to take some getting use to.

Oh wait... that's the wrong phrase. I should have said "That's going to take some with use to".